src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -122,7 +123,9 @@ async def pylint_lint_partition(partition: PylintPartition, pylint: Pylint) -> L
         PexRequest(
             output_filename="pylint.pex",
             internal_only=True,
-            requirements=PexRequirements([*pylint.all_requirements, *plugin_requirements]),
+            requirements=PexRequirements(
+                itertools.chain(pylint.all_requirements, plugin_requirements)
+            ),
```

Comment:
`itertools.chain` returns a generator. This avoids creating an unnecessary list in memory.

Reply:
This is an unnecessary optimization. This might matter if the list was huge, or if this was a performance-critical part of the code *and* you could prove that the cost of the import wasn't greater than the cost of the list construction. As it is I would be willing to bet that this actually slows things down due to the overhead of the import.